server/src/main/java/org/elasticsearch/index/query/CaseSensitivityMode.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.DeprecationHandler;
+
+import java.io.IOException;
+
+/** Case sensitivity matching mode for term-based queries */
+public enum CaseSensitivityMode implements Writeable {
+
+    /**
+     * use the field's setting for handling matching 
+     */
+    FIELD_DEFAULT(new ParseField("field_default")),
```

Comment:
I am wondering why we organize `case_sensitivity` options as parseFields,  why not just strings?
as `field_default` and `insensitive` are not field names, but rather field values?

Reply:
Thanks, Mayya. I was just following the pattern from `collect_mode` in aggs which uses an enum. This all may be academic because we might be going with a simple boolean, pending discussion (addressed in alternative PR https://github.com/elastic/elasticsearch/pull/63926 )